test/k8sT/Services.go

```diff
@@ -642,62 +713,105 @@ var _ = Describe("K8sServicesTest", func() {
+			res := kubectl.ExecInHostNetNS(context.TODO(), nodeName, cmd)
+			ExpectWithOffset(1, res).Should(helpers.CMDSuccess(),
+				"Cannot get IPv6 address for interface(%q): %s", iface, res.CombineOutput())
+			ipv6 := strings.Trim(res.Stdout(), "\n")
+
+			return ipv6
+		}
+
 		BeforeAll(func() {
 			demoYAML = helpers.ManifestGet(kubectl.BasePath(), "demo_ds.yaml")
-			res := kubectl.ApplyDefault(demoYAML)
-			res.ExpectSuccess("Unable to apply %s", demoYAML)
-			waitPodsDs()
-		})
 
-		AfterAll(func() {
-			// Explicitly ignore result of deletion of resources to avoid incomplete
-			// teardown if any step fails.
-			_ = kubectl.Delete(demoYAML)
-			ExpectAllPodsTerminated(kubectl)
-		})
+			DeployCiliumAndDNS(kubectl, ciliumFilename)
 
-		SkipItIf(helpers.RunsWithoutKubeProxy, "Checks ClusterIP Connectivity", func() {
-			service := "testds-service"
+			res := kubectl.ApplyDefault(demoYAML)
+			Expect(res).Should(helpers.CMDSuccess(), "Unable to apply %s", demoYAML)
 
-			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, service)
-			Expect(err).Should(BeNil(), "Cannot get service %s", service)
-			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+			if helpers.GetCurrentIntegration() == "" {
+				primaryK8s1IPv6 = getIPv6AddrForIface(k8s1NodeName, privateIface)
+				primaryK8s2IPv6 = getIPv6AddrForIface(k8s2NodeName, privateIface)
 
-			url := fmt.Sprintf("http://%s/", clusterIP)
-			testCurlFromPods(testDSClient, url, 10, 0)
+				// If there is no integration we assume that these are running in vagrant environment
+				// so have a secondary interface with both IPv6 and IPv4 addresses.
+				secondaryK8s1IPv4 = getIPv4AddrForIface(k8s1NodeName, helpers.SecondaryIface)
+				secondaryK8s2IPv4 = getIPv4AddrForIface(k8s2NodeName, helpers.SecondaryIface)
 
-			url = fmt.Sprintf("tftp://%s/hello", clusterIP)
-			testCurlFromPods(testDSClient, url, 10, 0)
-		})
+				secondaryK8s1IPv6 = getIPv6AddrForIface(k8s1NodeName, helpers.SecondaryIface)
+				secondaryK8s2IPv6 = getIPv6AddrForIface(k8s2NodeName, helpers.SecondaryIface)
 
-		SkipContextIf(helpers.RunsWithKubeProxy, "IPv6 Connectivity", func() {
-			testDSIPv6 := "fd03::310"
+				if helpers.ExistNodeWithoutCilium() {
+					outsideIPv6 = getIPv6AddrForIface(outsideNodeName, privateIface)
+				}
+			}
 
-			BeforeAll(func() {
-				// Install rules for testds-service (demo_ds.yaml)
-				httpBackends := ciliumIPv6Backends("-l k8s:zgroup=testDS,k8s:io.kubernetes.pod.namespace=default", "80")
-				ciliumAddService(31080, net.JoinHostPort(testDSIPv6, "80"), httpBackends, "ClusterIP", "Cluster")
-				tftpBackends := ciliumIPv6Backends("-l k8s:zgroup=testDS,k8s:io.kubernetes.pod.namespace=default", "69")
-				ciliumAddService(31069, net.JoinHostPort(testDSIPv6, "69"), tftpBackends, "ClusterIP", "Cluster")
-			})
+			if dualStackSupportEnabled {
```

Comment:
just search and replace `dualStackSupportEnabled` with `helpers.DualStackSupported()` :sweat_smile: